eng/common/scripts/Update-ChangeLog.ps1

```diff
@@ -12,18 +12,21 @@ param (
   [String]$ServiceDirectory,
   [Parameter(Mandatory = $true)]
   [String]$PackageName,
-  [boolean]$Unreleased=$True,
-  [boolean]$ReplaceLatestEntry = $False,
+  [String]$Unreleased=$True,
+  [String]$ReplaceLatestEntryTitle = $False,
   [String]$ReleaseDate
 )
 
-if ($ReleaseDate -and ($Unreleased -eq $True)) {
+[Boolean]$Unreleased = [System.Convert]::ToBoolean($Unreleased)
```

Comment:
Why do the manual conversion here as opposed to just having the parameters be Boolean as they currently are?

Reply:
We call it from various python and Js scripts. Passing the parameters a Booleans was problematic. It wasn't clear to me how to get it working from all ends with Boolean parameters hence I just did the conversion after.

Reply:
I figured you could just pass in the string "true" or "false"  (or 0/1) as arguments and it would just work. Did that not work when calling it from python/JS?

Reply:
Yes that the issue. Even thought "true" and "false" is passed in the [JS script ](https://github.com/Azure/azure-sdk-for-js/blob/0b736fcba541bc9640954f714b3af9b906cc9b2c/eng/tools/versioning/set-version.js#L69). It is still recognized a s string on the PowerShell side and errors if the parameters are Boolean. Python has a similar weirdness.

Reply:
OK. We could also consider making these switches so we don't have to pass a value in most cases. In either case if you keep these as strings it would be nice to add a comment as to why you are doing it so it is clear to use later.